euphonic/qpoint_frequencies.py

```diff
@@ -0,0 +1,240 @@
+import warnings
+from typing import Dict, Optional, TypeVar, Any
+
+import numpy as np
+
+from euphonic.validate import _check_constructor_inputs, _check_unit_conversion
+from euphonic.io import (_obj_to_json_file, _obj_from_json_file,
+                         _obj_to_dict, _process_dict)
+from euphonic.readers import castep, phonopy
+from euphonic.util import (_calc_abscissa, get_qpoint_labels)
+from euphonic import (ureg, Crystal, Quantity, Spectrum1D,
+                      Spectrum1DCollection)
+
+
+T = TypeVar('T', bound='QpointFrequencies')
+
+
+class QpointFrequencies(object):
+    """
+    A class to read and store frequency data at q-points
+
+    Attributes
+    ----------
+    crystal : Crystal
+        Lattice and atom information
+    n_qpts : int
+        Number of q-points in the object
+    qpts
+        Shape (n_qpts, 3) float ndarray. Q-point coordinates, in
+        fractional coordinates of the reciprocal lattice
+    frequencies
+        Shape (n_qpts, n_branches) float Quantity. Frequencies
+        per q-point and mode
+    weights
+        Shape (n_qpts,) float ndarray. The weight for each q-point
+    """
+
+    def __init__(self, crystal: Crystal, qpts: np.ndarray,
+                 frequencies: Quantity,
+                 weights: Optional[np.ndarray] = None) -> None:
+        """
+        Parameters
+        ----------
+        crystal
+            Lattice and atom information
+        qpts
+            Shape (n_qpts, 3) float ndarray. Q-point coordinates
+        frequencies
+            Shape (n_qpts, n_branches) float Quantity. Frequencies,
+            ordered according to increasing q-point number. Default
+            units meV
```

Comment:
What does it mean for a Quantity input to have 'default units meV'? Won't it be whatever units are given?

Reply:
This is probably left over from when Euphonic could only read from CASTEP files, to hint that the frequency units would be in meV. Removed now as its not generally applicable anymore.